src/event/quic/ngx_event_quic.c

```diff
@@ -1029,7 +1029,11 @@ ngx_quic_handle_payload(ngx_connection_t *c, ngx_quic_header_t *pkt)
          * After receiving a 1-RTT packet, servers MUST discard
          * 0-RTT keys within a short time
          */
-        ngx_quic_discard_ctx(c, ssl_encryption_early_data);
+        if (ngx_quic_keys_available(qc->keys, ssl_encryption_early_data, 0)
+            || ngx_quic_keys_available(qc->keys, ssl_encryption_early_data, 1))
```

Comment:
I feel like we can just call `ngx_quic_keys_discard()` without additional ifs. All necessary checks are present in the function itself.

Reply:
@arut thank you and I have addressed the review comment.

Reply:
also, `ngx_quic_keys_available(qc->keys, ssl_encryption_early_data, 1) is always false condition`